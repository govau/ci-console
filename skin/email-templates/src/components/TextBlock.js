import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Table, { Td } from "./Table";

const Block = styled(Table)`
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const BlockInner = styled(Td)`
  padding-top: 9px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const ContentContainer = styled(Table)`
  max-width: 100%;
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const Content = styled(Td)`
  padding-top: 0;
  padding-right: 18px;
  padding-bottom: 9px;
  padding-left: 18px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  word-break: break-word;
  color: ${({ theme, color }) => color || theme.textColor};
  font-family: Helvetica;
  line-height: 150%;
  font-family: ${theme => theme.fontFamily};
  text-align: ${({ align }) => align};
  font-size: ${({ fontSize }) => fontSize}px;
`;

const TextBlock = ({ children, align, color, fontSize, ...rest }) => {
  return (
    <Block border="0" cellPadding="0" cellSpacing="0" width="100%">
      <tbody>
        <tr>
          <BlockInner valign="top">
            <ContentContainer
              align="left"
              border="0"
              cellPadding="0"
              cellSpacing="0"
              width="100%"
            >
              <tbody>
                <tr>
                  <Content
                    align={align}
                    color={color}
                    fontSize={fontSize}
                    valign="top"
                  >
                    {children}
                  </Content>
                </tr>
              </tbody>
            </ContentContainer>
          </BlockInner>
        </tr>
      </tbody>
    </Block>
  );
};

TextBlock.propTypes = {
  align: PropTypes.oneOf(["left", "center", "right"]).isRequired,
  color: PropTypes.string,
  fontSize: PropTypes.number.isRequired
};

TextBlock.defaultProps = {
  align: "left",
  fontSize: 16
};

export default TextBlock;
