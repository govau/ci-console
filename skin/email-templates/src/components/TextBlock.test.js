import React from "react";
import renderer from "react-test-renderer";

import TextBlock from "./TextBlock";

test("renders without crashing", () => {
  const tree = renderer.create(<TextBlock>content</TextBlock>);
  expect(tree).toMatchSnapshot();
});
