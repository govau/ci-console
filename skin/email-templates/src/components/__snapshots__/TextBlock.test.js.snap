// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c1 {
  mso-table-lspace: 0pt;
  mso-table-rspace: 0pt;
}

.c0 {
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c2 {
  padding-top: 9px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c3 {
  max-width: 100%;
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c4 {
  padding-top: 0;
  padding-right: 18px;
  padding-bottom: 9px;
  padding-left: 18px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  word-break: break-word;
  font-family: Helvetica;
  line-height: 150%;
  text-align: left;
  font-size: 16px;
}

<table
  border="0"
  cellPadding="0"
  cellSpacing="0"
  className="c0 c1"
  width="100%"
>
  <tbody>
    <tr>
      <td
        className="c2"
        valign="top"
      >
        <table
          align="left"
          border="0"
          cellPadding="0"
          cellSpacing="0"
          className="c3 c1"
          width="100%"
        >
          <tbody>
            <tr>
              <td
                align="left"
                className="c4"
                color={undefined}
                fontSize={16}
                valign="top"
              >
                content
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
